classes/class-pmpro-subscription.php

```diff
@@ -654,16 +652,47 @@ public static function create( $args ) {
 	}
 
 	/**
-	 * Pull subscription info from the gateway.
+	 * Pull subscription info from the gateway or try to update
+	 * information from PMPro data.
 	 *
 	 * @since TBD
+	 *
+	 * @return bool True if the subscription was saved, false if not.
 	 */
-	public function update_from_gateway() {
-		$gateway_object = $this->get_gateway_object();
+	private function update() {
+		global $pmpro_level;
 
+		// Try to update directly from gateway.
+		$gateway_object = $this->get_gateway_object();
 		if ( $gateway_object && method_exists( $gateway_object, 'update_subscription_info' ) ) {
 			$gateway_object->update_subscription_info( $this );
+		} else {
+			// Update the start date to the date of the first order for this subscription.
+			$orders = $this->get_orders( [
+				'limit'   => 1,
+				'orderby' => '`timestamp` ASC, `id` ASC',
+			] );
+			if ( ! empty( $orders ) ) {
+				$order = current( $orders );
+				$this->startdate = date_i18n( 'Y-m-d H:i:s', $order->getTimestamp( true ) );
+			}
+
+			// Update the next payment date based on the most recent order.
+			if ( ! empty( $this->cycle_number ) ) {
+				// Only update the next payment date if we are not at checkout or there is no next payment date already set.
+				if ( ! isset( $pmpro_level ) || empty( $this->next_payment_date ) ) {
+					$orders = $this->get_orders( array( 'limit' => 1 ) );
```

Comment:
Should we reuse the $orders we received before at line 671?

Reply:
I don't know if we can. On 671, we want the oldest order to get the subscription startdate. On 684, we want to get the most recent order to guess at the next payment date.

Reply:
I guess we can grab them all and just pick out the first and last though

Reply:
If we rename `$orders` here to `$newest_orders` that would make it better for code readability and understanding intent too -- since we reuse `$orders` up above